src/main/java/apoc/map/Maps.java

```diff
@@ -178,18 +178,34 @@ public Object get(@Name("map") Map<String,Object> map, @Name("key") String key,
     }
 
     @UserFunction
-    @Description("apoc.map.removeKey(map,key)")
-    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key) {
+    @Description("apoc.map.removeKey(map,key,recursive:false) - remove the key from the map (recursively if recursive is true)")
+    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key, @Name(value="recursive", defaultValue = "false") boolean recursive) {
         Map<String, Object> res = new LinkedHashMap<>(map);
         res.remove(key);
+        if (recursive) {
+            for (Map.Entry<String, Object> entry : res.entrySet()) {
+                if (entry.getValue() instanceof Map) {
+                    Map<String, Object> updatedMap = removeKey((Map<String, Object>) entry.getValue(), key, true);
```

Comment:
here we can check if the updatedMap != entry.getValue() and only then (modified) update the value